pkg/kv/kvclient/kvcoord/txn_coord_sender.go

```diff
@@ -916,15 +916,15 @@ func (tc *TxnCoordSender) updateStateLocked(
 
 	// Update our transaction with any information the error has.
 	if errTxn := pErr.GetTxn(); errTxn != nil {
-		if errTxn.Status == roachpb.COMMITTED {
-			pErr = sanityCheckCommittedErr(ctx, pErr, ba)
+		if err := maybeSanityCheckCommittedErr(ctx, pErr, ba, &tc.testingKnobs); err != nil {
```

Comment:
nit: I think previous version was more readable because of explicit check for COMMITTED status prior to trying a sanity check. With new version I had to drill down into impl to understand why we feed all errors into commit check.

Reply:
interesting, the reason I pulled it in is because I had the opposite problem, that in the assertion we had to trust the caller that they had already checked the status. I felt that it needed to be either all in the method or completely inline, and I didn't want to inline it to keep the main code.

Were you thrown off by the name of the method referencing "committed"? We can rename it so that it's a general sanity check that.

Reply:
I think a better name would make sense. But I failed to come up with something sensible myself.